tests/test_dryrun.py

```diff
@@ -54,3 +68,55 @@ def test_get_referenced_tables(self, tmp_path):
         assert len(response) == 1
         assert response[0]["datasetId"] == "telemetry_derived"
         assert response[0]["tableId"] == "clients_daily_v6"
+
+    def test_get_referenced_tables_dml_stripped(self):
+
+        view_file = (
+            TEST_DIR / "sql" / "moz-fx-data-shared-prod" / "telemetry" / "view.sql"
+        )
+
+        dml_stripped = DryRun(sqlfile=str(view_file), strip_dml=True)
+        dml_unstripped = DryRun(str(view_file))
+
+        assert (
+            dml_stripped.get_referenced_tables()[0]["datasetId"] == "telemetry_derived"
+        )
+        assert dml_stripped.get_referenced_tables()[0]["tableId"] == "clients_daily_v6"
+        assert dml_unstripped.get_referenced_tables() == []
+
+    def test_get_referenced_tables_partition_filter(self):
+        """Test that DryRun returns referenced tables for
+        queries that require a partition filter."""
+
+        # this query requires a submission_date filter
+        view_file_1 = (
+            TEST_DIR
+            / "sql"
+            / "moz-fx-data-shared-prod"
+            / "messaging_system"
+            / "cfr_users_last_seen"
+            / "view.sql"
+        )
+
+        # this query requires a submission_timestamp filter
+        view_file_2 = (
+            TEST_DIR
+            / "sql"
+            / "moz-fx-data-shared-prod"
+            / "messaging_system"
+            / "onboarding_events_amplitude"
+            / "view.sql"
+        )
+
+        view_1 = DryRun(sqlfile=str(view_file_1), strip_dml=True)
+        view_2 = DryRun(sqlfile=str(view_file_2), strip_dml=True)
```

Comment:
I would test `get_error` directly here and ensure that you cover every enum value.